src/api.ts

```diff
@@ -1,206 +1,276 @@
+          {
+            writeonly: false,
+            readonly: true,
+            addReaonlyAndWriteonlyFilters: false
+          }
+          )
+        : null
+      if (returnType) {
+        writer.write(`this.axios.${analysis.operationMethod}<`)
+        writeWriterOrString(writer, returnType)
+        writer.write('>(\n')
+      } else {
+        writer.write(`this.axios.${analysis.operationMethod}(\n`)
+      }
+      // Endpoint
+      writer.indent(() => {
+        writer.quote(analysis.operationPath)
+        if (analysis.hasPathParams) {
+          // We need to replace url parameters in the endpoint
+          for (const param of analysis.pathParams) {
+            writer.write(`.replace(/{${param.name}}/, String(params[`).quote(param.name).write(']))')
+          }
+        }
+        // Data
+        if (analysis.isMethodWithData) {
+          if (analysis.hasRequestBody) {
+            writer.write(', data')
+          } else {
+            writer.write(', {}')
+          }
+        }
+        // Axios config
+        const needsAssign = analysis.hasHeaderParams || analysis.hasQueryParams
+        if (needsAssign) {
+          writer
+            .write(',')
+            .writeLine('Object.assign(')
+            .indent(() => {
+              writer
+                .writeLine('{},')
+                .inlineBlock(() => {
+                  if (analysis.hasHeaderParams) {
+                    writer.writeLine('headers: ' + CodeGen.generatePickString(analysis.headerParams) + ',')
+                  }
+                  if (analysis.hasQueryParams) {
+                    writer.writeLine('params: ' + CodeGen.generatePickString(analysis.queryParams) + ',')
+                  }
+                })
+                .write(',')
+                .writeLine('options')
+            })
+          writer.writeLine(')') // Assign ending paren
+        } else {
+          writer.write(', options')
+        }
+      })
+      writer.write(')') // Call ending paren
+    })
+  )
+}
+
+export async function generateApi (file: SourceFile, spec: OpenAPIV3.Document, opts: GenerateApiOptions): Promise<void> {
+  const genContext: GenerateApiContext = {
+    file,
+    spec,
+    options: opts
+  }
+
+  // Analyze all the operations in advance
+  const pathsAnalysis = Object.entries(spec.paths).map(
+    ([path, operations]) => analyzePathOperations(operations, { spec, path })
+  )
+
+  const referencedTypes = getAllReferencedTypes(pathsAnalysis)
 
-export async function generateApi (
-  file: SourceFile,
-  spec: OpenAPIV3.Document,
-  opts: { removeTagFromOperationId: boolean, addReadonlyWriteonlyModifiers: boolean }
-): Promise<void> {
   // Imports
   file.addImportDeclaration({
     defaultImport: 'axios',
     namedImports: ['AxiosInstance', 'AxiosRequestConfig'],
     moduleSpecifier: 'axios'
   })
-  file.addImportDeclaration({
-    namespaceImport: 'Types',
-    moduleSpecifier: './definitions'
-  })
+  if (referencedTypes.length) {
```

Comment:
You also need to import `WithoutWriteonly<>` and `WithoutReadonly<>` types if we need them (note: you also need to export them).